### PR TITLE
Chapter 8

### PR DESCRIPTION
**Pull request by damvanhoangbuu1**
Chapter 8: Basic login

### CODE REVIEW

**Comment by damvanhoangbuu1** [comment]
Ready

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class SessionsController < ApplicationController
+  def new; end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
```
Comment (vanvtt-0952) [inline app/controllers/sessions_controller.rb:6]: `user && user.authenticate(params[:session][:password])` thành `user&.authenticate(params[:session][:password])` nhé 

http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  end
+
+  def current_user
+    if session[:user_id]
+      @current_user ||= User.find_by(id: session[:user_id])
+    end
+  end
+
+  def logged_in?
+    !current_user.nil?
```
Comment (vanvtt-0952) [inline app/helpers/sessions_helper.rb:13]: `!current_user.nil?` thành `current_user.present?` nhé

**Comment by vanvtt-0952** [comment]
merged
2 comment nhỏ, fix rồi mợt đi nè

**Pull request merged**